clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -326,6 +340,7 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
             SelectionKey key = iterator.next();
             iterator.remove();
             KafkaChannel channel = channel(key);
+            long channelStartTimeNanos = recordTimePerConnection ? time.nanoseconds() : 0;
```

Comment:
I couldn't see any way around it. I will kick off system test runs to see the impact.